Jenkinsfile

```diff
@@ -1,57 +1,85 @@
+
+				// Install titanium SDK and select it
+				if ('GA'.equals(titaniumBranch)) {
+					sh 'ti sdk install latest -d'
+				} else if ('master'.equals(titaniumBranch)) {
+					sh 'ti sdk install -b master -d'
+				}
+				try {
+					withEnv(["PATH+ALLOY=${pwd()}/bin"]) {
+						sh 'npm test'
+					}
+				} finally {
+					junit 'TEST-*.xml'
+					stash includes: 'TEST-*.xml', name: "test-report-${titaniumBranch}"
+				}
+				
+				fingerprint 'package.json'
+			} // timeout
+		} // ansi
+	} // nodejs
+}
 
 timestamps() {
-	node('(osx || linux) && git && npm-publish && curl') {
-		def packageVersion = ''
-		def isPR = false
-		def published = false
-
-		stage('Checkout') {
-			// checkout scm
-			// Hack for JENKINS-37658 - see https://support.cloudbees.com/hc/en-us/articles/226122247-How-to-Customize-Checkout-for-Pipeline-Multibranch
-			checkout([
-				$class: 'GitSCM',
-				branches: scm.branches,
-				extensions: scm.extensions + [[$class: 'CleanBeforeCheckout'], [$class: 'CloneOption', honorRefspec: true, noTags: true, reference: '', shallow: true, depth: 30, timeout: 30]],
-				userRemoteConfigs: scm.userRemoteConfigs
-			])
-
-			isPR = env.BRANCH_NAME.startsWith('PR-')
-			packageVersion = jsonParse(readFile('package.json'))['version']
-			currentBuild.displayName = "#${packageVersion}-${currentBuild.number}"
-		}
+	try {
+		node('(osx || linux) && git ') {
+			stage('Lint') {
+				checkout scm
 
-		nodejs(nodeJSInstallationName: "node ${nodeVersion}") {
-			ansiColor('xterm') {
-				timeout(55) {
-					stage('Build') {
-						ensureNPM(npmVersion)
-						// Install yarn if not installed
+				packageVersion = jsonParse(readFile('package.json'))['version']
+				currentBuild.displayName = "#${packageVersion}-${currentBuild.number}"
+
+				nodejs(nodeJSInstallationName: "node ${nodeVersion}") {
+					ensureNPM(npmVersion)
+					// Install dependencies
+					timeout(5) {
 						sh 'npm ci'
-						if (sh(returnStatus: true, script: 'which ti') != 0) {
-							// Install titanium
-							sh 'npm install -g titanium'
-						}
-						if (sh(returnStatus: true, script: 'ti config sdk.selected') != 0) {
-							// Install titanium SDK and select it
-							sh 'ti sdk install -d'
-						}
-						try {
-							withEnv(["PATH+ALLOY=${pwd()}/bin"]) {
-								sh 'npm test'
-							}
-						} finally {
-							junit 'TEST-*.xml'
-						}
-						fingerprint 'package.json'
-					} // stage
-				} // timeout
+					}
+					sh 'npm run lint'
+				} // nodejs
+			} // stage('Lint')
+		} // node
 
-				stage('Security') {
-					// Clean up and install only production dependencies
+		stage('Test') {
+			parallel(
+				'GA SDK': {
+					node('(osx || linux) && git ') {
+						unitTests('GA', nodeVersion, npmVersion)
+					}
+				},
+				'master SDK': {
+					node('(osx || linux) && git ') {
+						unitTests('master', nodeVersion, npmVersion)
+
+					}
+				}
+			)
+		}
+
+		node('(osx || linux) && git ') {
+			stage('Security') {
+				checkout scm
```

Comment:
probably can get away with simply stashing the package.json/package-lock.json on initial check out and unstashing them here to avoid a full checkout for this.